src/verify.c

```diff
@@ -108,11 +117,44 @@ static void print_help(const char *name)
 	fprintf(stderr, "\n");
 }
 
+static bool compile_whitelist(const char *regex)
+{
+	int errcode;
+
+	if (picky_whitelist) {
+		regfree(picky_whitelist);
+	} else {
+		picky_whitelist = calloc(1, sizeof(*picky_whitelist));
```

Comment:
Then we are back to "statically allocated struct + extra variable", which is what I had before for the same reasons that you mention here. Okay, I'll go back to that, with a pointer instead of bool for the extra variable.